src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -425,9 +432,20 @@ private static AuthorityType GetAuthorityType(string authority)
                 return AuthorityType.B2C;
             }
 
+            if (isCiamAuthority(authority))
+            {
+                return AuthorityType.Ciam;
+            }
+
             return AuthorityType.Aad;
         }
 
+        private static bool isCiamAuthority(string authority)
+        {
+            var uri = new Uri(authority);
+            return uri.Host.Contains("ciamlogin.com");
```

Comment:
contains? or ends with?

Reply:
+1 to ends with.

Reply:
It should end with ".ciamlogin.com" (mind the dot at the beginning).